doc/examples/mnist.cpp

```diff
@@ -0,0 +1,219 @@
+/*******************************************************************************
```

Comment:
Might consider renaming this file and its associated `.hpp` to `mnist_loader` or something. For whatever reason, at first glance I was expecting to find an implementation of a model in here.